### ISSUE TITLE
“Report does not exist” in my dashboard

### ISSUE THREAD

**Issue by Hongbo-Miao** [stryker-mutator/stryker-dashboard#197]
**Summary**

Thanks for the awesome tool!

[My repo](https://github.com/Hongbo-Miao/hongbomiao.com) is a mono repo, I added Stryker to my `api` server folder at https://github.com/Hongbo-Miao/hongbomiao.com/blob/main/api/stryker.conf.js
Also, I added Stryker to my GitHub Actions pipeline, and it runs successfully.
https://github.com/Hongbo-Miao/hongbomiao.com/runs/2116804199?check_suite_focus=true

with log
```
...
22:31:43 (1715) INFO DashboardReporterClient PUT report to https://dashboard.stryker-mutator.io/api/reports/github.com/Hongbo-Miao/hongbomiao.com/PR-1972 (~976888 bytes)
22:31:43 (1715) INFO HtmlReporter Your report can be found at: file:///home/runner/work/hongbomiao.com/hongbomiao.com/api/reports/mutation/html/index.html
22:31:45 (1715) INFO DashboardReporter Report available at: https://dashboard.stryker-mutator.io/reports/github.com/Hongbo-Miao/hongbomiao.com/PR-1972
22:31:45 (1715) INFO MutationTestExecutor Done in 3 minutes 52 seconds.
```

You can see my pull request Stryker report
![image](https://user-images.githubusercontent.com/3375461/111819362-33bbcc80-891b-11eb-8a26-8d934e2402df.png)

However, my badge is not working,

![image](https://user-images.githubusercontent.com/3375461/111240022-5e650700-8635-11eb-9b52-1cd8b551de98.png)

Also, my dashboard is also not working.
https://dashboard.stryker-mutator.io/reports/github.com/Hongbo-Miao/hongbomiao.com/main

Does this issue is related with mono repo?

(I am also planning to add Stryker to my `web` and `mobile` app in this mono repo in future.)

**Stryker config**

```js
/**
 * @type {import('@stryker-mutator/api/core').StrykerOptions}
 */
module.exports = {
  packageManager: 'npm',
  reporters: ['html', 'clear-text', 'progress', 'dashboard'],
  testRunner: 'jest',
  coverageAnalysis: 'perTest',
};
```

**Test runner Jest config**

```js
module.exports = {
  testEnvironment: 'node',
  testPathIgnorePatterns: ['<rootDir>/build'],
};
```

**Stryker environment**

```
 "@stryker-mutator/core": "4.5.1",
 "@stryker-mutator/jest-runner": "4.5.1",
```

```
"jest": "26.6.3",
```

**Test runner environment**

```shell
npm test
```

```json
api/jest.config.js
```

**Your Environment**

| software         | version(s)
| ---------------- | -------
| node             | 14.16.0
| npm              | 7.6.3
| Operating System |

**Add stryker.log**

N/A. The test did run successfully in the GitHub Actions build which can be found above.

**Comment by nicojs** [comment]
Hi 👋 

Your dashboard seems to be working, see https://dashboard.stryker-mutator.io/reports/github.com/Hongbo-Miao/hongbomiao.com/PR-1972#mutant

The reason your `main` report doesn't exist is that it never seems to have been uploaded. 

The last part of the URL is what is referred to as the report's _version_. I.e.: In this URL: https://dashboard.stryker-mutator.io/reports/github.com/Hongbo-Miao/hongbomiao.com/PR-1972, the version is **PR-1972**. This URL refers to the **main** version: https://dashboard.stryker-mutator.io/reports/github.com/Hongbo-Miao/hongbomiao.com/main. Version roughly translates to the branch name or PR number on GitHub. 

Versions are essentially different reports. They don't interact with each other. Comparable to docker's "label" (or the label of an NPM package version "label").

The same principle works for the badge. Your badge seems to work fine for the PR-1972 report version:

![](https://img.shields.io/endpoint?url=https%3A%2F%2Fbadge-api.stryker-mutator.io%2Fgithub.com%2FHongbo-Miao%2Fhongbomiao.com%2FPR-1972)

(using this URL: https://img.shields.io/endpoint?url=https%3A%2F%2Fbadge-api.stryker-mutator.io%2Fgithub.com%2FHongbo-Miao%2Fhongbomiao.com%2FPR-1972)

In other words: you probably need to run the mutation testing task on your main branch.

**Comment by Hongbo-Miao** [comment]
Got it, thank you so much @nicojs ! 😃
